Assignment1/Assignment1/Controller/SuggestionViewController.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  SuggestionViewController.swift
+//  Assignment1
+//
+//  Created by Inito on 29/07/23.
+//
+
+import UIKit
+import Alamofire
+import FirebaseCore
+import FirebaseFirestore
+import FirebaseAuth
+
+class SuggestionViewController: UIViewController {
+
+    var items: [ [String : Any] ] = []
+    let db = Firestore.firestore()
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.dataSource = self
+        tableView.register(UINib(nibName: K.cellNibName, bundle: nil), forCellReuseIdentifier: K.cellIdentifier)
+        
+        let db = Firestore.firestore()
+        
+        
+        
+            self.loadProductData()
+        
+       
+       // fetchFilms()
+        // Do any additional setup after loading the view.
+    }
+    
+    func loadProductData(){
+        items = []
+        
+        db.collection(K.FStore.collectionName).getDocuments { (quearySnapshot, error) in
+            if let e = error {
+                print("ther is error in geting data \(e)")
+            }else{
+                if let snapShotDoc = quearySnapshot?.documents{
+                    var i = 0
+                    for doc in snapShotDoc {
+                        self.items.append(doc.data())
+                        DispatchQueue.main.async {
+                            self.tableView.reloadData()
+                        }
+
+                    }
+                }
+            }
+        }
+    }
+    
+    
+
+}
+
+
+extension SuggestionViewController: UITableViewDataSource{
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+      //  var totalRow = 0
+        //for i in 
+        return items.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let cell = tableView.dequeueReusableCell(withIdentifier: K.cellIdentifier, for: indexPath) as! ItemCell
+        
+        let item = items[indexPath.row]
+        var temp = item[K.FStore.itemInfo]
+
+        var infoItem = temp!
```

Comment:
don't use force unwrap

Reply:
done also change variable name "temp" to "item"